pkg/operator/resource/resourceapply/unstructured.go

```diff
@@ -0,0 +1,19 @@
+package resourceapply
+
+import (
+	"fmt"
+
+	"github.com/openshift/library-go/pkg/operator/events"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/client-go/dynamic"
+)
+
+// ApplyKnownUnstructured applies few selected Unstructured types, where it semantic knowledge
+// to merge existing & required objects intelligently. Feel free to add more.
+func ApplyKnownUnstructured(client dynamic.Interface, recorder events.Recorder, obj *unstructured.Unstructured) (*unstructured.Unstructured, bool, error) {
+	if obj.GetKind() == "ServiceMonitor" && obj.GetAPIVersion() == "monitoring.coreos.com/v1" {
```

Comment:
did `obj.GetObjectKind().GroupVersionKind().GroupKind()` not work?

Reply:
fixed